pkg/controllers/configuration.go

```diff
@@ -171,15 +203,38 @@ func (r *HypershiftDeploymentReconciler) ensureConfiguration(ctx context.Context
 		}
 
 		for _, se := range secretRefs {
-			k := genKey(se, hyd)
-			t, err := r.generateSecret(ctx, k, overrideNamespace(helper.GetHostingNamespace(hyd)))
+			// 1. Use user provided secret
+			k := genKey(se.secretRef, hyd)
+			secret, err := r.generateSecret(ctx, k, overrideNamespace(helper.GetHostingNamespace(hyd)))
 			if err != nil {
-				r.Log.Error(err, fmt.Sprintf("failed to copy secret %s", k))
+				r.Log.Info(fmt.Sprintf("failed to find and copy secret %s: %s", k, err.Error()))
+			}
+
+			if secret == nil {
+				// 2. Use existing secret in manifestwork payload
+				secret, err = getManifestPayloadSecretByName(&manifestwork.Spec.Workload.Manifests, se.secretRef.Name)
+				if err != nil {
+					r.Log.Info(fmt.Sprintf("failed to get etcd encryption key from manifestwork: %s", err.Error()))
```

Comment:
You generalized the generation of the secret, so you should not assume its the etcd encryption key. Just use the secret name.

Reply:
Also having Info and the word Fail, don't jive well. 😉